tests/ui/conditional-compilation/invalid-node-range-issue-129166.rs

```diff
@@ -0,0 +1,11 @@
+// This was triggering an assertion failure in `NodeRange::new`.
```

Comment:
Could you split the assert in `NodeRange::new` into two asserts?
Right now it is not clear which part of the condition actually fails.